### PR TITLE
refactor(auth)!: setLanguageCode no longer a setter

### LINKED ISSUE #3844: getters/settings not showing on docs website
## Documentation Feedback

E.g. https://rnfirebase.io/reference/auth - `languageCode` not showing.

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

### PR DESCRIPTION
**Pull request by russellwheatley**
### Description

- additional method for auth `setLanguageCode` so get & set ought to appear in documentation.
- updated test for `languageCode`

### Related issues

 - Fixes #3844

### Checklist

- I read the [Contributor Guide](/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [x] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/auth/lib/index.js [hunk 1]
```diff
@@ -129,6 +124,11 @@ class FirebaseAuthModule extends FirebaseModule {
     };
   }
 
+  async setLanguageCode(code) {
+    await this.native.setLanguageCode(code);
```
Comment (Ehesp) [inline packages/auth/lib/index.js:128]: This needs to check that code is a string or `null`.

File: packages/auth/lib/index.d.ts [hunk 2]
```diff
@@ -1207,7 +1192,19 @@ export namespace FirebaseAuthTypes {
      * > It is recommended to use {@link auth#onAuthStateChanged} to track whether the user is currently signed in.
      */
     currentUser: User | null;
-
+    /**
+     * Sets the language code.
```
Comment (Ehesp) [inline packages/auth/lib/index.d.ts:1196]: What happens on native if `null` is passed? Does it use the app language?

Comment (russellwheatley) [reply packages/auth/lib/index.d.ts:1196]: Android, errors when you pass `null` in: https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuth#public-void-setlanguagecode-string-languagecode

iOS doesn't error when you pass `null`. I can't actually find the documentation for the `FIRAuthRequestConfiguration` class that holds the language code value so I assume it is fairly old, I did find it in the iOS source code: https://github.com/firebase/firebase-ios-sdk/blob/42655733d51214ee83362afab0b0f545ce3b2102/FirebaseAuth/Sources/Backend/FIRAuthRequestConfiguration.h#L36 which confirms it is `nullable`. I've updated the code to handle both.

File: packages/auth/lib/index.d.ts [hunk 3]
```diff
@@ -1231,7 +1216,20 @@ export namespace FirebaseAuthTypes {
...
+     *
+     * ```js
+     * // Set language to French
+     * await firebase.auth().setLanguageCode('fr');
+     * ```
+     *
+     * @param code An ISO language code.
+     * 'null' value will set the language code to the app's current language.
+     */
+    setLanguageCode(languageCode: string | null);
```
Comment (mikehardy) [inline packages/auth/lib/index.d.ts:1232]:
```suggestion
    setLanguageCode(languageCode: string | null): Promise<void>;
```

**Comment by mikehardy** [comment]
@russellwheatley this seems like it could go but has conflicts from other parallel changes landing - if you fix and re-push, could be good ?

**Comment by russellwheatley** [comment]
@mikehardy done 😄  (Pending it passes flaky tests 🤞 )

**Comment by mikehardy** [comment]
@Ehesp this is blocked pending resolution of your request changes status

**Comment by mikehardy** [comment]
note to me or any potential merge-r: this is the one to carefully add a BREAKING CHANGES trailer note too so the release notes contain the brief-but-perfect instructions to upgraders on how to adopt

I was thinking

> BREAKING CHANGE: use setLanguageCode(), not direct property access, to set language code now

**Pull request merged**